openedx/core/djangoapps/content/search/documents.py

```diff
@@ -345,13 +347,10 @@ def _collections_for_content_object(object_id: OpaqueKey) -> dict:
             collections = authoring_api.get_entity_collections(
                 component.learning_package_id,
                 component.key,
-            )
+            ).values('key', 'title')
         elif isinstance(object_id, LibraryContainerLocator):
-            container = lib_api.get_container_from_key(object_id)
-            collections = authoring_api.get_entity_collections(
-                container.publishable_entity.learning_package_id,
-                container.key,
-            )
+            container = lib_api.get_container(object_id, include_collections=True)
+            collections = container.collections
```

Comment:
Not a change for this PR: but in the future, I think we should just refactor collections to be an independent API like tagging. Right now we have `get_container(object_id, include_collections=True)` and `container.collections` and `set_container_collections(...)` etc. as well as similar things for components. But why not just `get_collections(entity)`, `add_to_collection(entity)`, etc. that works with any object? I think that would be cleaner and simpler.